app/routes/calendar/new.tsx

```diff
@@ -331,15 +348,30 @@ function DatesInput() {
                       disabled={
                         datesCount === CALENDAR_EVENT.MAX_AMOUNT_OF_DATES
                       }
-                      onClick={() => setDatesCount((count) => count + 1)}
+                      onClick={() => {
+                        setDatesCount((count) => count + 1);
+                        setDatesInputState((current) => [
+                          ...current,
+                          {
+                            finalDateInputDate: getDateWithHoursOffset(
+                              // @ts-expect-error: this will never be null, so ignore this TSLint error
+                              datesInputState.at(i)?.finalDateInputDate
+                            ),
+                            index: i + 1,
+                          },
+                        ]);
```

Comment:
I think we can save us the trouble of always keeping two states in sync, by just deleting the `datesCount` state?
I don't see a reason to keep it, when we can just as well iterate over the values inside the `datesInputState` when rendering the date inputs, and will also get rid of the typescript error here, because we don't need to read it from a value 🙂 

(`datesCount === datesInputState.length`)

Reply:
Yes I agree & will refactor it when the time comes

Reply:
Done!